### ISSUE TITLE
ajax in sales

### ISSUE THREAD

**Issue by andriux1990** [opensourcepos/opensourcepos#2670]

**Comment by daN4cat** [comment]
@andriux1990, sorry I didn't find the time to catch up with this over the week.
Have you sorted? What's the status?

**Comment by andriux1990** [comment]
I decided to close the issue because I think nobody is interested. Well, I have it with Ajax, but I have the aforementioned problems and I like how it works if it weren't for this

**Comment by daN4cat** [comment]
I'll see if I find the time today to give a try and debug your issues.

**Comment by andriux1990** [comment]
OK thanks. I will upload the entire file now

**Comment by andriux1990** [comment]
[registerajax.zip](https://github.com/opensourcepos/opensourcepos/files/4050302/registerajax.zip)

**Comment by andriux1990** [comment]
toggle in ajax solved

**Comment by daN4cat** [comment]
Oh good, please send the code. I needed to fix other issues last weekend and didn't find the time to check your code yet.

**Comment by FAB123** [comment]
> toggle in ajax solved

I tested it. How I can fix my toggle

**Comment by andriux1990** [comment]
The problem I have now is that after adding a payment. And if I'm looking for another product to add for sale. The form_input is no longer displayed. I have to reload the page to show it.

If there are no aggregate payments, there is no problem, the entry form is always there and shows what is due correctly, but after adding a product using ajax it does not appear again until reloading the page

**Comment by daN4cat** [comment]
@andriux1990 have you managed to get this to work? Would be nice to have a register with ajax and no page refresh....

**Comment by andriux1990** [comment]
@daN4cat  Yes.. I made it work

**Comment by daN4cat** [comment]
@andriux1990 any chance to contribute it here?

**Comment by andriux1990** [comment]
Yes .. I'm not at home right now .. I'll share the file with you in a few hours

**Comment by daN4cat** [comment]
Thanks a lot!

**Comment by daN4cat** [comment]
@andriux1990 hope you haven't forgotten about this...

**Comment by andriux1990** [comment]
@daN4cat of course I have not forgotten

**Comment by andriux1990** [comment]
[registerajax.zip](https://github.com/opensourcepos/opensourcepos/files/5719474/registerajax.zip)

**Comment by daN4cat** [comment]
Thanks

**Comment by daN4cat** [comment]
@andriux1990 I cannot find the zip file, have you removed it?

**Comment by andriux1990** [comment]
@daN4cat I was able to make it now show the amount to pay after having added a payment, using ajax

https://user-images.githubusercontent.com/29256343/103394360-b176f580-4aed-11eb-9037-07f9f468910a.mp4

but it affects me when adding a client, I will solve this and share the file with you

**Comment by WebShells** [comment]
Hello @andriux1990 @daN4cat any updates on this ?

**Comment by andriux1990** [comment]
Hello @WebShells, how’s it going? I managed to fully implement AJAX in the sales module—100%, absolutely everything. The experience is really good.

However, even with all those improvements, it still didn’t reach the level I wanted. While it’s fast and the page doesn’t reload, it still felt lacking.

In the end, I stopped using the original sales module. Now I use one I developed myself, fully based on APIs and JavaScript, where all the sales logic is handled by the computer, and the system only receives the final data.
